src/energyplus/Test/GeometryTranslator_GTest.cpp

```diff
@@ -620,3 +626,470 @@ TEST_F(EnergyPlusFixture, GeometryTranslator_4ZoneWithShading_Simple_2) {
+
+Shading:Building,
+  Shading Building,        !- Name
+  0,                       !- Azimuth Angle {deg}
+  0,                       !- Tilt Angle {deg}
+  20,                      !- Starting X Coordinate {m}
+  0,                       !- Starting Y Coordinate {m}
+  5,                       !- Starting Z Coordinate {m}
+  10,                      !- Length {m}
+  5;                       !- Height {m}
+
+
+Shading:Building:Detailed,
+  Shading Building Detailed,  !- Detached Shading
+  ,                        !- Transmittance Schedule Name
+  4,                       !- Number of Vertices
+  10.0, 10.0, 5,            !- X,Y,Z ==> Vertex 1 {m}
+  10.0,  5.0, 5,            !- X,Y,Z ==> Vertex 2 {m}
+  20.0,  5.0, 5,            !- X,Y,Z ==> Vertex 3 {m}
+  20.0, 10.0, 5;            !- X,Y,Z ==> Vertex 4 {m}
+
+!-   ===========  ALL OBJECTS IN CLASS: CONSTRUCTION ===========
+
+Construction,
+  Wall,                    !- Name
+  Material 1;              !- Outside Layer
+
+Material,
+  Material 1,              !- Name
+  Rough,                   !- Roughness
+  0.00900,                 !- Thickness {m}
+  0.14000,                 !- Conductivity {W/m-K}
+  530.000,                 !- Density {kg/m3}
+  900.00,                  !- Specific Heat {J/kg-K}
+  0.900000,                !- Thermal Absorptance
+  0.600000,                !- Solar Absorptance
+  0.600000;                !- Visible Absorptance
+
+Output:Surfaces:Drawing,
+  DXF;                     !- Report Type
+    )IDF";
+
+  std::stringstream ss(testBuilding);
+  ss << "GlobalGeometryRules,\n"
+     << "  UpperLeftCorner,         !- Starting Vertex Position\n"
+     << "  CounterClockWise,        !- Vertex Entry Direction \n"
+     << "  " << coordinateSystem.valueName() << ", !- Coordinate System\n"
+     << "  " << coordinateSystem.valueName() << ", !- Daylighting Coordinate System\n"
+     << "  " << coordinateSystem.valueName() << "; !- Rectangular Surface Coordinate System\n"
+     << "\n"
+     << "Building,\n"
+     << "  Test building,           !- Name\n"
+     << "  " << northAxis << ",                      !- North Axis {deg}\n"
+     << "  Country,                 !- Terrain\n"
+     << "  0.04,                    !- Loads Convergence Tolerance Value {W}\n"
+     << "  0.4,                     !- Temperature Convergence Tolerance Value {deltaC}\n"
+     << "  FullInteriorAndExterior, !- Solar Distribution\n"
+     << "  ,                        !- Maximum Number of Warmup Days\n"
+     << "  6;                       !- Minimum Number of Warmup Days\n\n";
+
+
+  OptionalIdfFile oIdfFile = IdfFile::load(ss, IddFileType(IddFileType::EnergyPlus));
+  return Workspace(*oIdfFile);
+}
+
+
+enum class Quadrant {
+  LowerLeftCorner,
+  LowerRightCorner,
+  UpperRightCorner,
+  UpperLeftCorner
+};
+
+auto isSurfaceInQuadrant = [](const ShadingSurface& sc, Quadrant quadrant, const Transformation& expectedTransformation) {
```

Comment:
Helper to asserts whether a given surface is at the right position/rotation like we expect